src/components/VHeader/VHeaderInternal.vue

```diff
@@ -107,8 +99,11 @@ export default defineComponent({
       }
     })
 
-    watch(isMinScreenMd, (isMd) => {
-      if (isMd && isModalVisible.value) {
+    // We cannot use the `isDesktopLayout` from the ui store here because
+    // the threshold there is `lg`, and in `VHeaderInternal` it is `md`.
```

Comment:
`useLayout` checks for the feature flag, so shouldn't the UI store already be aware of this difference and compensating for it correctly? Maybe I'm not fully understanding how `useLayout` and the ui store are meant to work together.

Reply:
With the new header enabled, all the pages change the layout from mobile to desktop when they reach `lg` breakpoint. However, `VHeaderInternal`'s layout changes happens when the breakpoint is `md`. So, the `useLayout`'s `isDesktopLayout` will not be useful here (with the new header enabled) because the thresholds for desktop are different.
I wish I could write a clearer comment...

Reply:
Ahhh, I see! It was probably my bad assumption, I assumed that once the new header was enabled all the "desktop" breakpoints would be switching to the new one, I didn't realise it was fragmented in different parts of the page.

I wonder if we should just put the breakpoint itself into the cookie instead of doing the boolean `isDesktopLayout`, given "desktop layout" appears to potentially mean different things depending on which part of the view we're talking about.

My original intention in avoiding putting the breakpoint name into the cookie was to reduce the possibility for snooping on it to identify someone (moreso to increase user confidence than anything), but broad breakpoint range is not really that useful for fingerprinting anyway. Do you think it is worth the effort? Otherwise, I fear that anything that needs to know if we're in a specific breakpoint would be stuck not getting the benefits of the UI cookies for SSR.

Reply:
> I assumed that once the new header was enabled all the "desktop" breakpoints would be switching to the new one, I didn't realise it was fragmented in different parts of the page.

I assumed the same before looking closely at the mockups :) Here you can see the different thresholds:
<img width="285" alt="Screenshot 2022-11-16 at 10 18 33 AM" src="https://user-images.githubusercontent.com/15233243/202112353-9deeacf8-d882-4d59-85a7-159e0d3cf511.png">

> I wonder if we should just put the breakpoint itself into the cookie

I wanted to do that at first, as it would make having different thresholds for layouts easier.

The current implementation of the composables use `window.matchMedia`, so it does not return us the current breakpoint on screen change, it returns `true`/`false` when the mediaQuery like 'min-width: 640px' matches or doesn't match the current window.
To set the current breakpoint in the cookie, we would either have to use several `matchMedia` watchers, or watch `resize` event. I suspect that it would be less efficient than a single `matchMedia` event.

We also were going to discuss the pages and which ones of them are linked in the header/footer. I wonder if the threshold for the `hamburger` might change... @panchovm, is there such as possibility that we might have the `hamburger` menu for the page links up to the `lg` screen? Or is it certain that we will only have it until `md`?

Reply:
> is there such as possibility that we might have the hamburger menu for the page links up to the `lg` screen? Or is it certain that we will only have it until `md`?

My main concern with keeping the burger menu up to `lg` is that, on wider devices, the internal menu will look huge. Imagine this mockup on a 1024-px screen.

<img src="https://user-images.githubusercontent.com/895819/202142097-79cc6c28-a41f-4b1b-a19c-17a064cc1781.png" width="200">

On the other hand, I agree that we might need more space for showing links as the content needs are bigger than before.

I think we can keep the hamburger menu up to `lg` and improve the internal menu view once we define the pages listed. Some more in line with the current menu.

<img width="200" alt="CleanShot 2022-11-16 at 10 31 54@2x" src="https://user-images.githubusercontent.com/895819/202143367-d7f8a0ad-d03e-4536-8eeb-138b0eab1d6c.png">

Reply:
> To set the current breakpoint in the cookie, we would either have to use several matchMedia watchers, or watch resize event. I suspect that it would be less efficient than a single matchMedia event.

A single (debounced) resize event listener should not be significantly less efficient than a matchMedia event unless there's something really tricky going on between those two. At minimum, four matchMedia events to catch each relevant breakpoint wouldn't be worse than what we were doing before with multiple matchMedia events against the `md` breakpoints.

Of course both options are less efficient than a single matchMedia event, but I don't think significantly so (I could be wrong about that, though).

Given the trade-off here pretty minimal I'm not blocking this PR on this suggested change but I do think it would be a good ergonomic change to the cookie that would allow more flexibility like this into the future without much hassle if needed. Perhaps we can leave a note somewhere in the code of the UI store explaining this alternative approach and asking for it to be implemented if additional matchMedia events need to be added in the future (_en lieu_ of implementing it now, that is).

Reply:
> Perhaps we can leave a note somewhere in the code of the UI store explaining this alternative approach and asking for it to be implemented if additional matchMedia events need to be added in the future (en lieu of implementing it now, that is).

➕ to this, could one of you make a new issue for it?